R/create_query_data.R

```diff
@@ -346,14 +346,27 @@ get_terms_from_db <- function(version,
     queries = queries,
     i = i
   )
-  terms <- call_user_fun(
+
+  terms <- tryCatch(
     fun(
       basket_select = definition,
       version = version,
       keep_id = expect_grpid,
       temp_env = temp_env
-    )
+    ),
+    error = function(err) {
+      cli_abort(
+        c(
+          "An error occurred when calling the internal function:",
+          "Calling {.code fun(basket_select = {definition},
+          version = {version}, keep_id = {expect_grpid},
+          temp_env = {temp_env})} caused the following error:",
+          conditionMessage(err)
```

Comment:
Why "internal function"? We call a user provided function.
And the error message looks confusing:
```
> get_terms_from_db(
+         version = "test_version",
+         fun = error_fun,
+         queries = NULL,
+         definition = basket_select(name = "My basket", scope = "BROAD", type = "smq"),
+         expect_grpname = FALSE,
+         expect_grpid = FALSE,
+         i = 1,
+         temp_env = NULL
+       )
Error in `value[[3L]]()`:
! An error occurred when calling the internal function:
Calling `fun(basket_select = My basket, NULL, BROAD, and smq, version = test_version, keep_id = FALSE, temp_env = )`
caused the following error:
This is a simulated error.
```

Reply:
so we want the user to supply a custom function here?

it looked like originally you couldn't override the call_user_fun()

Reply:
The function provided via `get_terms_fun` in `create_query_data()` is called.

Reply:
so should it just be `...` instead of the function arguments? 
![image](https://github.com/user-attachments/assets/9b74c4a7-df6e-41a7-a689-8d297680d340)